mutation.hh

```diff
@@ -44,7 +44,7 @@ struct mutation_consume_cookie {
     bool partition_start_consumed = false;
     bool static_row_consumed = false;
     // only used when reverse == consume_in_reverse::yes
-    std::unique_ptr<range_tombstone_list> reversed_range_tombstones;
+    std::optional<range_tombstone> reversed_rt;
```

Comment:
This will become redundant with @tgrabiec work to store range tombstones in the clustering rows data structure. Doesn't hurt to include it, but is a waste of effort.